gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -103,9 +103,9 @@ message FeedEntity {
   optional TripUpdate trip_update = 3;
   optional VehiclePosition vehicle = 4;
   optional Alert alert = 5;
-
-  // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
   optional Shape shape = 6;
+  optional Stop stop = 7;
```

Comment:
Are the Stop FeedEntities required to be ordered before the TripModifications when combined in one feed? If not, is the consumer then required to filter, buffer, and reorder the messages it receives to ensure referential integrity (that new stops are known and created before being referenced by TripModifications?) How would this work in streaming differential cases?

Reply:
I don't think there's expectation of sorting (which is something that was present with shape in the past too). Yes it does mean the consumer needs to buffer the new entities. 

I also don't see how that's a problem for differential cases?

Reply:
The Stop is situated in a separate FeedEntity, outside the TripModification. Each FeedEntity contains exactly one sub-entity. In a message-oriented system transmitting individual differential updates, each containing one or more FeedEntities, how long should the receiver wait after receiving a TripModification that references a stop before it applies that update? When is the "safe" point to stop buffering and resolve the buffered/accumulated messages with respect to one another? If it receives a Stop message which alters an existing stop after it receives the TripModification, which version of the Stop does that TripModification reference? How will the receiver know whether a TripUpdate it has just received is intended to reference a stop that will be sent in a later message? How long is the sender allowed to wait before it supplies the Stops that the receiver is waiting for? At what point is it an error not to send Stops you've referenced? There are many questions like this that can likely be covered by a set of simple rules, but those rules would still need to be specified. 

One such rule is that messages may only reference entities created or modified in other messages that were sent before them. Another possible rule is that sending a message that updates an existing entity means all previously sent messages referencing that entity now reference the revised version of the entity (though this rule is not sufficient on its own to define what happens when the stop is not pre-existing).

Reply:
I agree those are good questions and pointers when `DIFFERENTIAL` is finalized. To me however this doesn't need to be finalized right now and can be defined later. Adding rules on how `DIFFERENTIAL` work when it's not specified anywhere else in the spec right doesn't make sense to me.